extensions/pkg/controller/infrastructure/controller.go

```diff
@@ -76,28 +77,32 @@ func DefaultPredicates(ignoreOperationAnnotation bool) []predicate.Predicate {
 
 // Add creates a new Infrastructure Controller and adds it to the Manager.
 // and Start it when the Manager is Started.
-func Add(mgr manager.Manager, args AddArgs) error {
+func Add(mgr manager.Manager, args AddArgs, predicates []predicate.Predicate) error {
 	args.ControllerOptions.Reconciler = NewReconciler(mgr, args.Actuator)
-	return add(mgr, args)
+	return add(mgr, args, predicates)
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, args AddArgs) error {
+func add(mgr manager.Manager, args AddArgs, predicates []predicate.Predicate) error {
 	ctrl, err := controller.New(ControllerName, mgr, args.ControllerOptions)
 	if err != nil {
 		return err
 	}
 
-	predicates := extensionspredicate.AddTypePredicate(args.Predicates, args.Type)
+	p := extensionspredicate.AddTypePredicate(predicates, args.Type)
 
-	if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Infrastructure{}}, &handler.EnqueueRequestForObject{}, predicates...); err != nil {
+	if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Infrastructure{}}, &handler.EnqueueRequestForObject{}, p...); err != nil {
 		return err
 	}
 
-	if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Cluster{}}, &extensionshandler.EnqueueRequestsFromMapFunc{
-		ToRequests: extensionshandler.SimpleMapper(ClusterToInfrastructureMapper(predicates), extensionshandler.UpdateWithNew),
-	}); err != nil {
-		return err
+	// do not watch cluster if respect operation annotation to prevent unwanted reconciliations in case the operation annotation
+	// is already present & the extension CRD is already deleting
+	if args.IgnoreOperationAnnotation {
+		if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Cluster{}}, &extensionshandler.EnqueueRequestsFromMapFunc{
```

Comment:
When I was discussing the issue with @timuthy, he shared the following scenario:
```
Suppose you have a cluster in deletion and after some time it transitions to Failed  state. Then the op or user can trigger a retry which resets the Failed  state, right? Therefore, the deletion must continue somehow but it’s not triggered by operation: reconcile  because it’s in deletion
```

Does this path from above work when we remove the Cluster watch with this PR?

Reply:
Yes, because we wanted to add the timestamp as annotation also for deletion (which will trigger the controller and make it enqueueing the extension object again for deletion), see https://github.com/gardener/gardener/pull/2290#issuecomment-624627927.

Reply:
Will add it to the deletion too.